### PR TITLE
[Shipping Labels Revamp] Update Carrier and Saved packages data for better state control

### LINKED ISSUE #12747: [Shipping Labels: Packages] Add Package Opening UI Section
Summary
==========
Once the Package UI flow is functional, we should allow it to be actioned from the main Shipping Labels creation flow, this task focus on this feature by introducing an opening view within the main form.

Designs
==========
Image
Image

### PR DESCRIPTION
**Pull request by ThomazFB**
Summary
==========
Partially fix issue #12747 by introducing a series of adjustments to the Carrier and Saved packages data to allow a complete state control with loading and error handling.

The upcoming PR will be focused on triggering the selected package to the main Shipping Labels UI.

Screenshots
==========
https://github.com/user-attachments/assets/3e50c713-66ca-468b-b2d5-cba735535704

How to Test
==========
1. Go to the Order Details of an Order eligible for Shipping Label Creation
2. Hit the Create Shipping Labels button
3. Once inside the new Shipping Labels form, hit the `Select a Package` button
4. Verify that the Carrier and Saved tabs display their loading feedback properly
5. Verify that the `Add Package` button is only enabled for the tab with an actual selected Package
6. Verify that the Custom Package form logic is working as expected

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by atorresveiga** [comment]
Good work @ThomazFB ! I checked that:

✅  The Carrier and Saved tabs display their loading feedback properly
✅  The Add Package button is only enabled for the tab with an actual selected Package
✅  The Custom Package form logic is working as expected

Some notes from my testing not related to the changes in this PR:

- Seems that vertical scroll is missing on some of the screens. I tried to scroll down on the Carrier screen with no luck and look at this weird behaviour when creating a custom package

https://github.com/user-attachments/assets/58dd96ee-5d44-4723-bece-900f6c0f7d6a

Adding a custom package works, but we need to add some feedback (loading state)

**Review by atorresveiga — APPROVED** [review]
LGTM! :shipit:

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/packages/ui/PackageDataTest.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package com.woocommerce.android.ui.orders.wooshippinglabels.packages.ui
+
+import org.assertj.core.api.Assertions.assertThat
+import kotlin.test.Test
+
+class PackageDataTest {
```
Comment (atorresveiga) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/packages/ui/PackageDataTest.kt:6]: Thanks!

**Comment by ThomazFB** [comment]
Hey @atorresveiga! Thank you so much for testing 🙇 Regarding the scroll, I actually fixed that yesterday but noticed that the fix went with the [other PR](https://github.com/woocommerce/woocommerce-android/pull/13111) 🤦 
image

To avoid doing git cherry picks or complicating things, I'll leave this to be fixed in that PR since the adjustment is already there, WDYT? 😬 

>Adding a custom package works, but we need to add some feedback (loading state)

I agree with your point! There's one thing, though. When I left the Custom package creation without loading feedback, the reasoning was how we could avoid making the user wait for the package to be saved. We could be "optimistic" about the request and just move forward with the Shipping Labels form. 

With that, when we hit the `Add Package` button, the data will be sent immediately to both the Shipping Labels form and the networking layer, making the experience feel snap and removing the need for the loading feedback. WDYT?

**Comment by ThomazFB** [comment]
@atorresveiga I'll move forward with this PR merge to unblock the next one, and depending on what we decide from the discussion here, I can introduce some fixes through the next PR. Hope that's ok!

**Comment by atorresveiga** [comment]
> With that, when we hit the Add Package button, the data will be sent immediately to both the Shipping Labels form and the networking layer, making the experience feel snap and removing the need for the loading feedback. WDYT?

My only concern is what will happen if the save request fails? If we move to the shipping label form, we cannot recover from that, and the package will not be saved. WDYT?

**Pull request merged**